pkg/cluster/status.go

```diff
@@ -0,0 +1,88 @@
+package cluster
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sretry "k8s.io/client-go/util/retry"
+
+	v2 "github.com/percona/percona-server-mysql-operator/pkg/api/v2"
+	"github.com/percona/percona-server-mysql-operator/pkg/database/mysql"
+	"github.com/percona/percona-server-mysql-operator/pkg/database/orchestrator"
+)
+
+type component interface {
+	GetSize() int32
+	MatchLabels() map[string]string
+}
+
+func (r *MySQLReconciler) updateStatus(cr *v2.PerconaServerForMySQL) error {
+	m := mysql.New(cr)
+	mysqlStatus, err := r.appStatus(cr, m)
+	if err != nil {
+		return errors.Wrap(err, "get MySQL status")
+	}
+	cr.Status.MySQL = mysqlStatus
+
+	o := orchestrator.New(cr)
+	orcStatus, err := r.appStatus(cr, o)
+	if err != nil {
+		return errors.Wrap(err, "get Orchestrator status")
+	}
+	cr.Status.Orchestrator = orcStatus
+
+	return r.writeStatus(cr)
```

Comment:
I don't feel like we calculate the whole status correctly here.
e.g. is CR ready if all pods are running, but MySQL pods are not added to Orchestrator or when no Primary at this moment, etc.

Reply:
Actually, we don't provide CR status yet. We just update status for `orchestrator` and `mysql` if they reached the target size.